sdk/src/sharedTest/java/io/customer/sdk/repository/ProfileRepositoryTest.kt

```diff
@@ -189,11 +214,33 @@ class ProfileRepositoryTest : BaseTest() {
     }
 
     @Test
-    fun clearIdentify_givenNoPreviouslyIdentifiedProfile_expectIgnoreRequest_expectDontDeleteDeviceToken() {
+    fun clearIdentify_givenNoPreviouslyIdentifiedProfile_expectIgnoreRequest_expectDontDeleteDeviceToken_expectDontUpdateHook() {
         repository.clearIdentify()
 
         prefRepository.getIdentifier().shouldBeNull()
+
         verify(deviceRepositoryMock, never()).deleteDeviceToken()
+        verify(hooksManager, never()).onHookUpdate(any())
+    }
+
+    @Test
+    fun clearIdentify_givenPreviouslyIdentifiedProfile_expectHookUpdate() {
+        val givenIdentifier = String.random
+        prefRepository.saveIdentifier(givenIdentifier)
+
+        repository.clearIdentify()
+
+        prefRepository.getIdentifier().shouldBeNull()
+
+        val argumentCaptor =
+            nullableArgumentCaptor<ModuleHook.BeforeProfileStoppedBeingIdentified>()
+
+        verify(hooksManager, times(1)).onHookUpdate(
+            argumentCaptor.capture() ?: ModuleHook.BeforeProfileStoppedBeingIdentified(
```

Comment:
I don't understand the reason for `?: ModuleHook.BeforeProfileStoppedBeingIdentified(`? 

Are we sure it's a good idea to give a default value here? Shouldn't we test when `onHookUpdate()` gets called with `null`?

Reply:
updated it, we don't need it anymore, as `ModuleHook` doesn't need to be nullable.